ksql-engine/src/main/java/io/confluent/ksql/function/udaf/sum/SumUdaf.java

```diff
@@ -0,0 +1,201 @@
+        0,
+        Integer::sum,
+        (valueToUndo, aggregateValue) -> aggregateValue - valueToUndo
+    );
+  }
+
+  @UdafFactory(description = "Sum double values.")
+  public static TableUdaf<Double, Double, Double> sumDouble() {
+
+    return getSumImplementation(
+        0.0,
+        Double::sum,
+        (valueToUndo, aggregateValue) -> aggregateValue - valueToUndo
+    );
+  }
+
+  @UdafFactory(description = "Sum double values in a list into a single double")
+  public static TableUdaf<List<Long>, Long, Long> sumLongList() {
+    return getListSumImplementation(
+        0L,
+        (list) -> sumLongList(list),
+        Long::sum,
+        (valueToUndo, aggregateValue) -> aggregateValue - valueToUndo
+    );
+  }
+
+  private static long sumLongList(final List<Long> list) {
+    return sumGenericList(list, 0L, (a,b) -> a + b);
+  }
+
+  @UdafFactory(description = "Sum integer values in a list into a single integer")
+  public static TableUdaf<List<Integer>, Integer, Integer> sumIntList() {
+    return getListSumImplementation(
+        0,
+        (list) -> sumIntList(list),
+        Integer::sum,
+        (valueToUndo, aggregateValue) -> aggregateValue - valueToUndo
+    );
+  }
+
+  private static int sumIntList(final List<Integer> list) {
+    return sumGenericList(list, 0, (a,b) -> a + b);
+  }
+
+  @UdafFactory(description = "Sum double values in a list into a single double")
+  public static TableUdaf<List<Double>, Double, Double> sumDoubleList() {
+    return getListSumImplementation(
+        0.0,
+        (list) -> sumDoubleList(list),
+        Double::sum,
+        (valueToUndo, aggregateValue) -> aggregateValue - valueToUndo
+    );
+  }
+
+  private static double sumDoubleList(final List<Double> list) {
+    return sumGenericList(list, 0.0, (a,b) -> a + b);
+  }
+
+  private static  <I> TableUdaf<I, I, I> getSumImplementation(
+      final I initialValue,
+      final BiFunction<I, I, I> summer,
+      final BiFunction<I, I, I> undoer) {
+
+    return new TableUdaf<I, I, I>() {
+
+      @Override
+      public I initialize() {
+        return initialValue;
+      }
+
+      @Override
+      public I aggregate(final I newValue, final I aggregateValue) {
+        if (newValue == null) {
+          return aggregateValue;
+        }
+        return summer.apply(newValue, aggregateValue);
+      }
+
+      @Override
+      public I map(final I aggregate) {
+        return aggregate;
+      }
+
+      @Override
+      public I merge(final I agg1, final I agg2) {
+        return summer.apply(agg1, agg2);
+      }
+
+      @Override
+      public I undo(final I valueToUndo, final I aggregateValue) {
+        if (valueToUndo == null) {
+          return aggregateValue;
+        }
+        return undoer.apply(valueToUndo, aggregateValue);
+      }
+    };
+  }
+
+  private static <I> TableUdaf<List<I>, I, I> getListSumImplementation(
+      final I initialValue,
+      final Function<List<I>, I> sumList,
```

Comment:
You don't need this as a separate parameter.  You have all the information you need to sum the list -> you have the `initialValue` and the `summer`, so you can use these with `sumGenericList` to replace `sumList`, e.g. `aggregate` becomes:

```java
      @Override
      public I aggregate(final List<I> newValue, final I aggregateValue) {
        if (newValue == null) {
          return aggregateValue;
        }
       final I total = sumGenericList(newValue, initialValue, summer)
        return summer.apply(aggregateValue, total);
      }
```